### PR TITLE
honoring config option `force_ip_to_sysname` in bill creation and editing

### PR DESCRIPTION
**Pull request by TvL2386**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: includes/html/modal/new_bill.inc.php [hunk 1]
```diff
@@ -44,12 +44,18 @@
...
-                                echo "<option value='${device['device_id']}' $selected>${device['hostname']}</option>\n";
-                            }
-                                ?>
+                              $sort_field = 'hostname';
+
+                              if (\LibreNMS\Config::get('force_ip_to_sysname') === true) {
+                                  $display_field = 'sysName';
+                              }
+
+                              $devices = dbFetchRows('SELECT * FROM `devices` ORDER BY ?', array($display_field));
```
Comment (PipoCanaja) [inline includes/html/modal/new_bill.inc.php]: Hi @TvL2386 
Thanx for your PR ! It seems that there are still some variable to rename. 
:)

Comment (TvL2386) [reply includes/html/modal/new_bill.inc.php]: Thanks PipoCanaja, I feel ashamed...

**Comment by PipoCanaja** [comment]
Now, following CodeClimate and Travis suggestions will allow you to correct the indent issues and get this PR approuved.

File: includes/html/modal/new_bill.inc.php [hunk 2]
```diff
@@ -44,12 +44,18 @@
...
-                                echo "<option value='${device['device_id']}' $selected>${device['hostname']}</option>\n";
-                            }
-                                ?>
+                              $sort_field = 'hostname';
+
+                              if (\LibreNMS\Config::get('force_ip_to_sysname') === true) {
+                                  $sort_field = 'sysName';
+                              }
+
+                              $devices = dbFetchRows('SELECT * FROM `devices` ORDER BY ?', array($sort_field));
```
Comment (murrant) [inline includes/html/modal/new_bill.inc.php]: This actually won't sort correctly if the user mixes dns named and ip hosts.

You need to fetch the devices then sort them in PHP.

Honestly, you should use select2 for these.  Search for init_select2 in the code to see how.

Comment (laf) [reply includes/html/modal/new_bill.inc.php]: > This actually won't sort correctly if the user mixes dns named and ip hosts.

That's how it is now though so this doesn't regress anything just honours an existing config option.

Comment (TvL2386) [reply includes/html/modal/new_bill.inc.php]: I thought so too laf.

Comment (murrant) [reply includes/html/modal/new_bill.inc.php]: Yeah, but if you use the select2 code, the device names will respect the setting, be sorted properly and not blow up if there are 100,000 devices or something.

Yes, code is fine as is.

Really all you need is this: 
```php
<select class="form-control input-sm" id="device" name="device" onchange="getInterfaceList(this)"></select>
<script type="text/javascript">
    init_select2('#device', 'device', {}, <?php echo "{id: $port_device_id, text: '" . format_hostname($device) . "'}");
</script>
```

(You'll have to get $device from somewhere to find the hostname)

**Comment by SourceDoctor** [comment]
@TvL2386 
could you resolve merge conflicts, please?

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-63-release-changelog-april-2020/11828/1

**Pull request merged**